File: src/main/java/com/server/ordering/domain/Review.java
package com.server.ordering.domain;

import lombok.*;

import javax.persistence.*;

import static javax.persistence.FetchType.*;
import static javax.persistence.GenerationType.*;
import static lombok.AccessLevel.*;

@Entity
@Getter
@RequiredArgsConstructor
@NoArgsConstructor(access = PROTECTED)
public class Review {

    @Id @GeneratedValue(strategy = IDENTITY)
    @Column(name = "review_id")
    private Long id;

    @NonNull
    private String review;

    @NonNull
    @Column(nullable = false)
    private Float rating;

    private String imageUrl;

    @ManyToOne(fetch = LAZY)
    @JoinColumn(name = "restaurant_id")
    private Restaurant restaurant;

    @OneToOne(fetch = LAZY)
    @JoinColumn(name = "order_id")
    private Order order;

    public void registerOrderAndRestaurant(Order order, Restaurant restaurant) {
        this.order = order;
        this.restaurant = restaurant;
        order.registerReview(this);
        restaurant.registerReview(this);
    }

    public void putReview(String text, float rating) {
        this.review = text;
        this.rating = rating;
    }

    public void registerImageUrl(String imageUrl) {
        this.imageUrl = imageUrl;
    }
}

File: src/main/kotlin/com/byeoru/ordering_server/domain/Review.kt
package com.byeoru.ordering_server.domain

import javax.persistence.*

@Entity
class Review(review: String,
             rating: Float) {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "review_id")
     var id: Long? = null
         protected set

     var review: String = review
         protected set

    @Column(nullable = false)
     var rating: Float = rating
         protected set

     var imageUrl: String? = null
         protected set

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "restaurant_id")
    private var restaurant: Restaurant? = null

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "order_id")
     lateinit var order: Order
         protected set

    fun registerOrderAndRestaurant(order: Order, restaurant: Restaurant) {
        this.order = order
        this.restaurant = restaurant
        order.registerReview(this)
        restaurant.registerReview(this)
    }

    fun putReview(text: String, rating: Float) {
        review = text
        this.rating = rating
    }

    fun registerImageUrl(imageUrl: String?) {
        this.imageUrl = imageUrl
    }
}